File: backend/src/models/query.py
"""Query execution Pydantic models."""

from typing import Any

from pydantic import Field

from src.utils.camel_case import CamelCaseModel


class QueryRequest(CamelCaseModel):
    """Request model for SQL query execution."""

    sql: str = Field(..., description="SQL query to execute")


class QueryResult(CamelCaseModel):
    """Response model for SQL query execution."""

    columns: list[str] = Field(..., description="Column names")
    rows: list[dict[str, Any]] = Field(..., description="Result rows")
    row_count: int = Field(..., description="Number of rows returned")
    execution_time: float = Field(..., description="Execution time in milliseconds")


class NaturalLanguageQueryRequest(CamelCaseModel):
    """Request model for natural language query."""

    prompt: str = Field(..., description="Natural language query prompt")


class NaturalLanguageQueryResponse(CamelCaseModel):
    """Response model for natural language query."""

    sql: str = Field(..., description="Generated SQL query")
    explanation: str = Field(..., description="Explanation of the generated SQL")

File: frontend/src/types/query.ts
/**
 * TypeScript types for query execution.
 */

export interface QueryRequest {
  sql: string;
}

export interface QueryResult {
  columns: string[];
  rows: Record<string, any>[];
  rowCount: number;
  executionTime: number; // milliseconds
}

export interface NaturalLanguageQueryRequest {
  prompt: string;
}

export interface NaturalLanguageQueryResponse {
  sql: string;
  explanation: string;
}
